src/lib/firewall.ts

```diff
@@ -191,16 +195,20 @@ export async function applyFirewallMode(mode: string) {
 								target: 'REJECT',
 							}),
 					)
-					.forChain('INPUT', (chain) =>
-						chain
-							.addRule({
-								action: iptables.RuleAction.Flush,
-							})
-							.addRule({
-								action: iptables.RuleAction.Append,
+					.forChain('INPUT', (chain) => {
+						if (firewallRuleExists) {
+							chain.addRule({
+								action: iptables.RuleAction.Delete,
 								target: 'BALENA-FIREWALL',
-							}),
-					),
+							});
+						}
+						chain.addRule({
+							action: iptables.RuleAction.Insert,
```

Comment:
Does the rule have to go in the same position? Why?

Reply:
Could we alternatively just remove the existing rule and add a new one each time, what are the drawbacks?

Reply:
The change does remove the existing rule and add a new one. It maintains the same position to preserve iptables behavior when there are multiple rules in the INPUT chain. since the rules evaluate based on their position in the chain.

Reply:
I agree that we need to preserve the order, so that we do not scramble the the rules if there is some change from a container that would be affected by reordering.